### PR TITLE
Check and uncheck

### PR DESCRIPTION
**Pull request by ahohnsen**

### CODE REVIEW

File: src/components/TodoItem.spec.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  });
+
+  it('can be checked and unchecked', () => {
+    const {rerender} = render(<TodoItem id="1" description="Buy chocolate" isDone={false} />);
+
+    expect(screen.getByRole('checkbox')).not.toBeChecked();
+
+    rerender(<TodoItem id="1" description="Buy chocolate" isDone={true} />);
+    expect(screen.getByRole('checkbox')).toBeChecked();
+  });
```
Comment (beetlejays) [inline src/components/TodoItem.spec.js:30]: Hätte ich genauso geschrieben ;)

File: src/components/TodoItem.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  return (
+    <StyledListItem>
+      <input type="checkbox" onChange={() => onToggle()} checked={isDone} id={id} />
+      <label htmlFor={id}>{description}</label>
+    </StyledListItem>
+  );
+}
+
+const StyledListItem = styled.li`
+  word-wrap: anywhere;
```
Comment (smilenow24) [inline src/components/TodoItem.js:13]: i like word-wrap

File: src/App.js [hunk 3]
```diff
@@ -10,15 +12,26 @@ export default function App() {
...
   );
 
   function addTodo(description) {
-    setTodos([...todos, description]);
+    setTodos([...todos, {id: nanoid(), description: description, isDone: false}]);
+  }
+
+  function toggleTodo(index) {
+    const todo = todos[index];
+    setTodos([...todos.slice(0, index), {...todo, isDone: !todo.isDone}, ...todos.slice(index + 1)]);
```
Comment (GeorgJuranek) [inline src/App.js:34]: nice slice-techniqie

File: src/components/TodoItem.spec.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+    const user = userEvent.setup();
+    await user.click(screen.getByLabelText('Buy chocolate'));
+    expect(callback).toHaveBeenCalled();
+  });
+
+  it('can be checked and unchecked', () => {
+    const {rerender} = render(<TodoItem id="1" description="Buy chocolate" isDone={false} />);
+
+    expect(screen.getByRole('checkbox')).not.toBeChecked();
```
Comment (bensle) [inline src/components/TodoItem.spec.js:26]: Thanks for explanation to "getByRole"

**Review by bensle — APPROVED** [review]

File: src/components/TodoItem.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  list-style-type: none;
+
+  input:checked + label {
+    text-decoration: line-through;
+    color: hotpink;
```
Comment (isabellpaulmann) [inline src/components/TodoItem.js:18]: Tolle Farbwahl!

**Review by GeorgJuranek — COMMENTED** [review]
good work

File: src/components/TodoItem.js (hunk 2 above)
Comment (ahohnsen) [reply src/components/TodoItem.js:13]: Thank you :)

File: src/components/TodoItem.spec.js (hunk 1 above)
Comment (ahohnsen) [reply src/components/TodoItem.spec.js:30]: Thanks!

File: src/components/TodoItem.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    <StyledListItem>
+      <input type="checkbox" onChange={() => onToggle()} checked={isDone} id={id} />
+      <label htmlFor={id}>{description}</label>
+    </StyledListItem>
+  );
+}
+
+const StyledListItem = styled.li`
+  word-wrap: anywhere;
+  list-style-type: none;
```
Comment (sarina-nikolic) [inline src/components/TodoItem.js:14]: Isn't "list-style" enough?

File: src/components/TodoItem.js (hunk 5 above)
Comment (lindajk) [inline src/components/TodoItem.js:18]: why hotpink??????

**Review by lindajk — APPROVED** [review]

**Comment by HuiminFrontend** [comment]
Thank you for explanation.

File: src/components/TodoItem.js (hunk 5 above)
Comment (ahohnsen) [reply src/components/TodoItem.js:18]: because it's Jerry's favorite :D

File: src/components/TodoItem.spec.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
+import {render, screen} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import TodoItem from './TodoItem.js';
+
+describe('TodoItem', () => {
```
Comment (RicardoBachmann) [inline src/components/TodoItem.spec.js:6]: good to know how the follow steps are working ;-)

File: src/components/TodoItem.spec.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    expect(todoItem).toBeInTheDocument();
+  });
+
+  it('calls onToggle on click', async () => {
+    const callback = jest.fn();
+    render(<TodoItem id="1" description="Buy chocolate" isDone={false} onToggle={callback} />);
+
+    const user = userEvent.setup();
+    await user.click(screen.getByLabelText('Buy chocolate'));
+    expect(callback).toHaveBeenCalled();
```
Comment (Zaki5581) [inline src/components/TodoItem.spec.js:20]: review

**Pull request merged**